File: client/Client/DTOs/MapDTO.cs
using LoadRunnerClient.MapAndModel;

namespace LoadRunnerClient.DTOs
{
	/// <summary>
	/// Equivalent of Server-Side DTO to pass a Map
	/// </summary>
	public class MapDTO
    {
        private MapMetaDTO _meta;
        private Tile[][] _tiles;
        private Item[][] _items;
		private EnemySpawnPointDTO[] _enemySpawnPoints;
        private Vector2Int _spawn;

        public MapDTO()
        {
        }

		/// <summary>
		/// Constructor of the MapDTO
		/// </summary>
		/// <param name="meta">MapMetaDTO instance providing information about properties of the Map</param>
		/// <param name="tiles">Collection of Tile Enums describing the Tiles Type</param>
		/// <param name="items">Collection of Item Enums describing the Items Type</param>
		public MapDTO(MapMetaDTO meta, Tile[][] tiles, Item[][] items)
        {
            this._meta = meta;
            this._tiles = tiles;
            this._items = items;
        }

        public MapMetaDTO meta
        {
            get => _meta;
            set => _meta = value;
        }

        public Tile[][] tiles
        {
            get => _tiles;
            set => _tiles = value;
        }


        public Item[][] items
        {
            get => _items;
            set => _items = value;
        }
    
		public EnemySpawnPointDTO[] enemySpawnPoints {
			get => _enemySpawnPoints;
			set => _enemySpawnPoints = value;
		}

        public Vector2Int spawn
        {
            get => _spawn;
            set => _spawn = value;
        }

	}
}

File: server/src/main/java/de/hsrm/mi/swtpro03/server/game/dtos/MapDTO.java
package de.hsrm.mi.swtpro03.server.game.dtos;

import de.hsrm.mi.swtpro03.server.game.model.Item;
import de.hsrm.mi.swtpro03.server.game.model.Map;
import de.hsrm.mi.swtpro03.server.game.model.Tile;
import de.hsrm.mi.swtpro03.server.game.model.Vector2;
import de.hsrm.mi.swtpro03.server.game.model.Vector2Int;

import java.util.Arrays;
import java.util.List;

public class MapDTO {

	private MapMetaDTO meta;
	private Tile[][] tiles;
	private Item[][] items;
	private EnemySpawnPointDTO[] enemySpawnPoints;
	private Vector2Int spawn;



	// Default Constructor for Jackson
	public MapDTO() {
	}

	public MapDTO(MapMetaDTO meta, Tile[][] tiles, Item[][] items, Vector2Int spawn, EnemySpawnPointDTO[] enemySpawnPoints) {
		this.meta = meta;
		this.tiles = tiles;
		this.items = items;
		this.enemySpawnPoints = enemySpawnPoints;
		this.spawn = spawn;
	}

	public MapMetaDTO getMeta() {
		return meta;
	}

	public void setMeta(MapMetaDTO meta) {
		this.meta = meta;
	}

	public Tile[][] getTiles() {
		return tiles;
	}

	public void setTiles(Tile[][] tiles) {
		this.tiles = tiles;
	}

	public Item[][] getItems() {
		return items;
	}

	public void setItems(Item[][] items) {
		this.items = items;
	}	

	public EnemySpawnPointDTO[] getEnemySpawnPoints() {
		return enemySpawnPoints;
	}

	public void setEnemySpawnPoints(EnemySpawnPointDTO[] enemySpawnPoints) {
		this.enemySpawnPoints = enemySpawnPoints;
	}

	public Vector2Int getSpawn() {
		return spawn;
	}

	public void setSpawn(Vector2Int spawn) {
		this.spawn = spawn;
	}

	public Map toMap() {
		Map map = new Map(meta.getNumberOfSections(), meta.getName(), spawn);
		for (int x = 0; x < meta.getNumberOfSections() * Map.SECTION_WIDTH; x++) {
			for (int y = 0; y < Map.SECTION_HEIGHT; y++) {
				map.addTile(x, y, tiles[x][y]);
				map.addItem(x, y, items[x][y]);
			}
		}

		if (enemySpawnPoints != null) {
			for (EnemySpawnPointDTO enemySpawnPoint : enemySpawnPoints) {
				map.addEnemySpawnPoint(enemySpawnPoint);
			}
		}
		return map;
	}

	@Override
	public boolean equals(Object obj) {
		if (obj instanceof MapDTO) {
			MapDTO dto = (MapDTO) obj;
			return dto.meta.equals(this.meta)
					&& Arrays.deepEquals(dto.tiles, this.tiles)
					&& Arrays.deepEquals(dto.items, this.items)
					&& Arrays.deepEquals(dto.enemySpawnPoints, this.enemySpawnPoints);
		}
		return false;
	}
}
